### PR TITLE
[bug] fix #2003 unsigned transaction count upon expiration

### LINKED ISSUE #2003: Wallet: Unsigned header shows wrong number of transactions after transaction is expired
### Description

*Type*: Bug

*Summary*: New Wallet -> Transactions -> Unsigned 
header shows number of transactions in the list, e.g `Unsigned 1`. After transaction is expired and removed from the list the header is not updated and shows old count number. Expected: update the header, so 
1. if there are no transactions should be `Unsigned`
2. if there are some other transactions left then it should show the corresponding number, e.g 2 transactions are in the list and 1 is just expired -> header is `Unsigned 2`
image 1

### Reproduction
See from 05:21 https://app.testfairy.com/projects/4803590-status/builds/6722465/sessions/8/?accessToken=-NJ26YzdeceCQ5H4tDwntRo2gU8
- Open Status
- Open new Wallet (tap on tab Wallet)
- Send some ether to Jarrad but do not sign transaction and select Sign Later
- After transaction is moved to Unsigned -> open Unsigned tab and wait 5 minutes (for tx to time out)
- Check the header after transaction is deleted from the list because of time out

### Additional Information
* Status version: develop 0.9.11d163 
* Operating System: 
Real device iPhone 7 plus, iOs 10.3.2

### PR DESCRIPTION
**Pull request by yenda**
fixing #2003

Before:
screenshot_1507207147
After:
screenshot_1507263339
screenshot_1507263366

### Summary:

Fix color of transaction count and refresh when transactions expire

this fixed involved changing the swiper so that a function can be passed as title instead of a string.
this allows developpers to pass custom elements as titles and still use style and variables that are computed when tab is built

edit: completely reworked the swiper

### Steps to test:
- Open Status
- Send a transaction but sign later
- Go to unsigned tab and wait 5 min for the transaction to expire
- Count and transaction should have disappeared

[comment]: # (PRs will only be accepted if squashed into single commit.)
status: ready

### CODE REVIEW

File: src/status_im/components/styles.cljs [hunk 1]
```diff
@@ -25,6 +25,7 @@
 (def color-gray7 "#9fa3b4")
 (def color-gray8 "#6E777E")
 (def color-gray9 "#E9EBEC")
+(def color-gray10 "#9BA3A8")
```
Comment (jeluard) [inline src/status_im/components/styles.cljs:28]: Will it ever stop ?? ;)

File: src/status_im/components/tabs/views.cljs [hunk 2]
```diff
@@ -9,20 +9,25 @@
             [status-im.components.tabs.styles :as tabs.styles]
             [status-im.utils.platform :as platform]))
 
-(defn- tab [{:keys [view-id title icon-active icon-inactive style-active selected-view-id on-press]}]
+(defn- tab [{:keys [view-id title title-fn icon-active icon-inactive style-active selected-view-id on-press]
+             :or {title-fn react/text}}]
```
Comment (jeluard) [inline src/status_im/components/tabs/views.cljs]: Smart!

File: src/status_im/components/tabs/views.cljs [hunk 3]
```diff
@@ -9,20 +9,25 @@
...
      [react/view {:style tabs.styles/tab-container}
       (when-let [icon (if active? icon-active icon-inactive)]
         [react/view
          [vi/icon icon (tabs.styles/tab-icon active?)]])
-      [react/view
-       [react/text (merge (if text-only? {:uppercase? (get-in platform/platform-specific [:uppercase?])})
-                          {:style (tabs.styles/tab-title active? text-only?)})
-        title]]]]))
+      [react/view {:style tabs.styles/tab-title-wrapper}
+       (title-fn {:uppercase? (and text-only? (get-in platform/platform-specific [:uppercase?]))
```
Comment (jeluard) [inline src/status_im/components/tabs/views.cljs]: Not exactly the same here as `uppercase?` as no general meaning, only for text-only component.
Probably harmless.

File: src/status_im/ui/screens/wallet/transactions/views.cljs [hunk 4]
```diff
@@ -131,17 +131,25 @@
...
-(defn- unsigned-transactions-title [unsigned-transactions-count]
-  (str (i18n/label :t/transactions-unsigned)
-       (if (pos? unsigned-transactions-count) (str " " unsigned-transactions-count))))
+(defn unsigned-transactions-title [{:keys [uppercase? style]} title]
+  (let [unsigned-transactions-count (re-frame/subscribe [:wallet.transactions/unsigned-transactions-count])]
+    [react/view {:flex-direction :row}
+     [react/text {:style style
+                  :uppercase? uppercase?}
+      (i18n/label :t/transactions-unsigned)]
+     (when (pos? @unsigned-transactions-count)
```
Comment (jeluard) [inline src/status_im/ui/screens/wallet/transactions/views.cljs]: `@` coud be called only once

File: src/status_im/ui/screens/wallet/transactions/views.cljs [hunk 5]
```diff
@@ -131,17 +131,25 @@
                       :render-fn       render-transaction
                       :empty-component (empty-text (i18n/label :t/transactions-unsigned-empty))}]]))
 
-(defn- unsigned-transactions-title [unsigned-transactions-count]
-  (str (i18n/label :t/transactions-unsigned)
-       (if (pos? unsigned-transactions-count) (str " " unsigned-transactions-count))))
+(defn unsigned-transactions-title [{:keys [uppercase? style]} title]
```
Comment (jeluard) [inline src/status_im/ui/screens/wallet/transactions/views.cljs]: I don't think `uppercase?` and `style` should be managed by wrapping component as those are specific to this particular implementation.
If the idea was to share behavior between default `title-fn` and this one I'd rather we use functions and duplicate the definition of properties.

**Review by jeluard — CHANGES_REQUESTED** [review]

File: src/status_im/components/styles.cljs (hunk 1 above)
Comment (flexsurfer) [reply src/status_im/components/styles.cljs:28]: Designers never sleep!

File: src/status_im/ui/screens/wallet/transactions/views.cljs [hunk 6]
```diff
@@ -131,17 +131,25 @@
...
+  (let [unsigned-transactions-count (re-frame/subscribe [:wallet.transactions/unsigned-transactions-count])]
```
Comment (flexsurfer) [inline src/status_im/ui/screens/wallet/transactions/views.cljs]: better to use `defview`, and don't use re-frame/subscribe at all

**Review by flexsurfer — APPROVED** [review]

**Review by jeluard — APPROVED** [review]

**Pull request merged**